### PR TITLE
Support for a ZK Certificate Registry with a Queue

### PR DESCRIPTION
**Pull request by jilio**
```mermaid
sequenceDiagram
    participant User as User
    participant CLI as CLI
    participant Blockchain as Blockchain
    participant Registry as Registry
    participant MerkleService as Merkle Proof Service
    participant GuardianRegistry as Guardian Registry

    User ->> CLI: Run issueZKCert command
    CLI ->> Blockchain: Connect to RPC
    CLI ->> GuardianRegistry: Verify provider is a guardian
    GuardianRegistry -->> CLI: Confirmation
    CLI ->> MerkleService: Get empty leaf proof
    MerkleService -->> CLI: Empty leaf proof
    CLI ->> Registry: Register to queue
    Registry -->> CLI: Queue registration confirmation
    CLI ->> Blockchain: Wait for queue time
    Blockchain -->> CLI: Queue time available
    CLI ->> Registry: Add zkCertificate to registry
    Registry -->> CLI: Certificate addition confirmation
    CLI ->> User: Output transaction details and save certificate

    note over CLI: Certificate issued successfully
```

### CODE REVIEW

File: pkg/zkcertificate/certificate.go [hunk 1]
```diff
@@ -191,8 +197,8 @@ func SignCertificate(
 		return nil, fmt.Errorf("hash message: %w", err)
 	}
 
-	// TODO: Why mod here? It doesn't crash without mod
-	// message = message.Mod(message, utils.NewIntFromString("2736030358979909402780800718157159386076813972158567259200215660948447373040"))
+	// take modulo of the message to get it into the mod field supported by eddsa
+	message = message.Mod(message, utils.NewIntFromString(eddsaPrimeFieldMod))
```
Comment (HarmlessEvil) [inline pkg/zkcertificate/certificate.go:201]: I feel like we should extract hash function that would call `poseidon.Hash` and take this modulo, so we won't forget it anywhere else

Comment (HarmlessEvil) [inline pkg/zkcertificate/certificate.go:201]: Also I think it'd be better to call `utils.NewIntFromString(eddsaPrimeFieldMod)` once in the initialisation of the global variable

File: cmd/issueZKCert.go [hunk 2]
```diff
@@ -310,3 +309,81 @@ func encodeMerkleProof(proof merkle.Proof) [][32]byte {
...
+		currentTime := big.NewInt(time.Now().Unix())
+		if startTime.Cmp(currentTime) <= 0 {
+			break
+		}
+		if expirationTime.Cmp(currentTime) <= 0 {
+			return fmt.Errorf("queue wait time expired for zkCertificate hash %s", leafHash)
+		}
+
+		select {
+		case <-time.After(5 * time.Second):
```
Comment (HarmlessEvil) [inline cmd/issueZKCert.go:382]: Maybe we should wait until `min(startTime, now() + 5 seconds)`?

Comment (jilio) [reply cmd/issueZKCert.go:382]: It is safe as expiration time is a few minutes.

**Review by HarmlessEvil — APPROVED** [review]
I don't like a name of the function, but good enough. Thank you!

**Pull request merged**